Framework/API/src/ExperimentInfo.cpp

```diff
@@ -302,31 +304,17 @@ void ExperimentInfo::setInstrument(const Instrument_const_sptr &instr) {
   }
   const auto parInstrument = Geometry::ParComponentFactory::createInstrument(
       sptr_instrument, m_parmap);
+
+  m_infoVisitor = makeOrRetrieveVisitor(*sptr_instrument);
```

Comment:
Could this alleviate the performance regression that @VickieLynch observed? Currently, if the same instrument without visitor is set on multiple `ExperimentInfos`, `Instrument::registerContents` will be called multiple times?

On the other hand, what you are doing right now does not seem to do more work than what we had before, so I do not understand why there is a slowdown?